### PR TITLE
Allow editing files when editing posts

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Allow user to add and remove files when editing a post.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-62011

#### Release Note
```release-note
none
```

### CODE REVIEW

File: webapp/channels/src/actions/post_actions.ts [hunk 1]
```diff
@@ -323,9 +324,16 @@ export function unpinPost(postId: string): ActionFuncAsync<boolean> {
...
 
 export function setEditingPost(postId = '', refocusId = '', isRHS = false): ActionFunc<boolean, GlobalState> {
+    const getFilesForPost = makeGetFilesForPost();
+
     return (dispatch, getState) => {
         const state = getState();
-        const post = PostSelectors.getPost(state, postId);
+        let post = PostSelectors.getPost(state, postId);
+
+        if (post.file_ids?.length) {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/actions/post_actions.ts:333]: lets add a comment here for clarification

File: webapp/channels/src/components/post/post_component.test.tsx [hunk 2]
```diff
@@ -324,4 +324,139 @@ describe('PostComponent', () => {
...
+            };
+
+            const props = {
+                ...baseProps,
+                post,
+            };
+
+            const {container} = renderWithContext(<PostComponent {...props}/>, state);
+            expect(screen.getByTestId('fileAttachmentList')).toBeInTheDocument();
+            expect(container.querySelectorAll('.post-image__column')).toHaveLength(3);
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/post/post_component.test.tsx:363]: is `querySelectorAll` not available in screen?

**Review by M-ZubairAhmed — APPROVED** [review]
No blocking just few questions

File: webapp/channels/src/actions/post_actions.ts (hunk 1 above)
Comment (harshilsharma63) [reply webapp/channels/src/actions/post_actions.ts:333]: Done.

File: webapp/channels/src/components/post/post_component.test.tsx (hunk 2 above)
Comment (harshilsharma63) [reply webapp/channels/src/components/post/post_component.test.tsx:363]: No, screen only has the selectors for text content, test id etc. The more powerful selectors are in `container`.

**Pull request merged**